Keas.Mvc/Services/IdentityService.cs

```diff
@@ -187,7 +187,7 @@ public async Task<User> GetByKerberos(string kerb)
                 FirstName = ucdKerbPerson.DFirstName,
                 LastName = ucdKerbPerson.DLastName,
                 Id = ucdKerbPerson.UserId,
-                Email = ucdContact.Email,
+                Email = ucdContact.Email ?? (ucdKerbPerson.UserId != null ? $"{ucdKerbPerson.UserId}@ucdavis.edu" : null),
```

Comment:
Just use an IF statement here, nesting a null coalescence `??` and a tertiary together is not easy to read.  `var email = ucdContext.Email;  if (email == null) { ... }` that kind of thing so we can tell what's happening